openstates/nm/legislators.py

```diff
@@ -28,101 +28,74 @@ def scrape_legislator(self, chamber, term, url):
-                id_ = 'ctl00_mainCopy_formViewLegislator_%s' % value
-                val = None
-                try:
-                    val = "\n".join(doc.get_element_by_id(id_).itertext())
-                    found = True
-                except KeyError:
-                    pass
-                if val:
-                    properties[key] = val.strip()
-                else:
-                    properties[key] = None
-
-            if found is False and key not in optional:
-                self.warning('bad legislator page %s missing %s' %
-                             (url, id_))
-                return
-
-        # image & email are a bit different
-        properties['photo_url'] = doc.xpath('//img[@id="ctl00_mainCopy_formViewLegislator_imgLegislator"]/@src')[0]
-        email = doc.get_element_by_id('ctl00_mainCopy_formViewLegislator_linkEmail').text
-        
-
-        properties['url'] = url
-
-        properties['chamber'] = chamber
-        properties['term'] = term
+        xpath = '//span[@id="ctl00_mainCopy_formViewLegislator_{}"]/text()'
 
-        full_name, party = properties['header'].rsplit("-", 1)
+        district = doc.xpath('//a[@id="ctl00_mainCopy_formViewLegislator_linkDistrict"]/text()')[0]
 
-        properties['full_name'] = full_name.replace("Representative","").replace("Senator","").strip()
-        properties['party'] = party
+        header = doc.xpath(xpath.format('lblHeader'))
+        full_name, party = header[0].rsplit("-", 1)
+        full_name = full_name.replace("Representative","").replace("Senator","").strip()
 
-        if '(D)' in properties['party']:
-            properties['party'] = 'Democratic'
-        elif '(R)' in properties['party']:
-            properties['party'] = 'Republican'
-        elif '(DTS)' in properties['party']:
+        if '(D)' in party:
+            party = 'Democratic'
+        elif '(R)' in party:
+            party = 'Republican'
+        elif '(DTS)' in party:
             # decline to state = independent
-            properties['party'] = 'Independent'
+            party = 'Independent'
         else:
             raise Exception("unknown party encountered")
 
-        address = properties.pop('addr_street')
-
-        phone = (properties.pop('office_phone') or
-                 properties.pop('home_phone'))
-
-        leg = Legislator(**properties)
-        leg.add_source(url)
+        photo_url = doc.xpath('//img[@id="ctl00_mainCopy_formViewLegislator_imgLegislator"]/@src')[0]
 
+        email = doc.xpath('//a[@id="ctl00_mainCopy_formViewLegislator_linkEmail"]/text()')
         if email:
-            properties['email'] = email.strip()
+            email = email[0]
         else:
             print "no email"
             email = None
 
-        leg.add_office('district', 'District Address', address=address,
-                       phone=phone, email=email)
+        district_address = ", ".join(doc.xpath(xpath.format('lblAddress')))
```

Comment:
Separate the street address from the city-state-ZIP using a `newline` instead of a comma.